services/commonService.js

```diff
@@ -0,0 +1,82 @@
+fluid.registerNamespace("myL3");
+
+fluid.defaults("myL3.commonService", {
+    gradeNames: "fluid.component",
+    invokers: {
+        getData: {
+            funcName: "myL3.commonService.getData",
+            args: ["{arguments}.0"]
+        }
+    }
+});
+
+myL3.commonService.getData = (fields) => {
+
+        if($('#tabularData tbody').children().length > 1) {
+            $('#weatherTable').hide();
+            const children = $('#tabularData tbody').children();
+            for(let i=1; i<children.length; i++) {
+                $('#tabularData tbody').children()[1].remove();
+            }
+            myL3.values = [];
+            myL3.weatherService.addDataIntoTable(myL3.oneTimeWeatherData, fields);
+            return;
+        }
+
+        if (navigator.geolocation) {
+
+            options = {
+                enableHighAccuracy: true,
+                timeout: Infinity,
+                maximumAge: 0
+            };
+
+            error = (err) => {
+                console.log(err);
+                $('#err').html(`Cannot retrieve location info, \
+                <br>Reason: <strong>${err.message}</strong>`);
+                if(!navigator.onLine) {
+                    alert('Internet connection not avaiable');
+                }
+            }
+
+            getWeatherData = (position) => {
+                myL3.oneTimePosition = position;
+                console.log('Coordinates: ', position);
+                const lat = position.coords.latitude;
+                const lon = position.coords.longitude;
+
+                $.getJSON(`https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&APPID=${myL3.apiKeys.openweathermapKey}`, (weatherData) => {
```

Comment:
Please split the service integration code from the UI code. As we discussed, a class is a good structure for the service integration.

The code making the AJAX call should also not use `myL3.apiKeys.openweathermapKey` directly -- better to make it into a parameter or class instance variable that is provided to the code from its client. In this way, the code that is responsible for making the service call is independent from the storage of the API keys. The API key is something that the service code needs, but the API key is given by the client, rather than the service code being responsible for getting it.

Reply:
```
var mapKey = prompt("Please enter Google maps key:");
    script.src = `https://maps.googleapis.com/maps/api/js?key=${mapKey}&callback=myMap`;
```
Will it work in case of API keys

In the browser it look something like this:
![image](https://user-images.githubusercontent.com/26577556/41924517-e7351fa6-7987-11e8-85c1-a99a312d3554.png)